### PR TITLE
Issue1059

### PR DESCRIPTION
**Pull request by roeger**

### CODE REVIEW

**Review by jendrikseipp — COMMENTED** [review]
Some minor nitpicks :-)

**Comment by silvansievers** [comment]
I would like to comment on `find_unsatisfied_condition` in `stubborn_sets.h`. As this function wasn't changed, I cannot comment on it in the pull request. Anyway, I wondered if this function has to be defined in the header and declared `inline`. I haven't seen this in a long way and I think usually we declare the function in the header, as `extern`, and then implement it in the implementation file. I'm not sure if the compiler won't still be able to inline it, if this is a performance concern.

File: src/search/pruning/stubborn_sets.h [hunk 1]
```diff
@@ -9,14 +9,6 @@ inline FactPair find_unsatisfied_condition(
     const std::vector<FactPair> &conditions, const State &state);
```
Comment (silvansievers) [inline src/search/pruning/stubborn_sets.h:9]: I think the function `find_unsatisfied_condition` could be moved to the action-centric class. The atom-centric only uses the `find_unsatisfied_precondition` function. (Note the subtle difference.)

File: src/search/pruning/stubborn_sets_action_centric.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    */
+    FactPair find_unsatisfied_goal(const State &state) const {
+        return find_unsatisfied_condition(sorted_goals, state);
+    }
+
+    // Return true iff the operator was enqueued.
+    // TODO: rename to enqueue_stubborn_operator?
+    bool mark_as_stubborn(int op_no);
+    virtual void compute_stubborn_set(const State &state) override;
+    virtual void initialize_stubborn_set(const State &state) = 0;
```
Comment (silvansievers) [inline src/search/pruning/stubborn_sets_action_centric.h:34]: I think both `initialize_stubborn_set` and `handle_stubborn_operator` could be private.

Comment (roeger) [reply src/search/pruning/stubborn_sets_action_centric.h:34]: Done.

File: src/search/DownwardFiles.cmake [hunk 3]
```diff
@@ -335,6 +335,14 @@ fast_downward_plugin(
     DEPENDENCY_ONLY
 )
 
+fast_downward_plugin(
+    NAME STUBBORN_SETS_ACTION_CENTRIC
+    HELP "Base class for all action-centric stubborn set partial order reduction methods"
+    SOURCES
+        pruning/stubborn_sets_action_centric
+    DEPENDS STUBBORN_SETS
```
[the remark this answers is not in the document]
Comment (roeger) [reply src/search/DownwardFiles.cmake:343]: Done.

File: src/search/pruning/stubborn_sets.h (hunk 1 above)
Comment (roeger) [reply src/search/pruning/stubborn_sets.h:9]: The implementation of `find_unsatisfied_precondition` uses `find_unsatisfied_condition`.

Comment (silvansievers) [reply src/search/pruning/stubborn_sets.h:9]: Right. My IDE didn't show me this.

**Review by silvansievers — COMMENTED** [review]
This commit lacks the "[issue1059] " prefix. (Resolved)

File: src/search/pruning/stubborn_sets_action_centric.cc [hunk 4]
```diff
@@ -0,0 +1,62 @@
+#include "stubborn_sets_action_centric.h"
+
+using namespace std;
+
+namespace stubborn_sets {
+StubbornSetsActionCentric::StubbornSetsActionCentric(const options::Options &opts)
+    : StubbornSets(opts) {
+}
```
Comment (silvansievers) [inline src/search/pruning/stubborn_sets_action_centric.cc:8]: newline

File: src/search/pruning/stubborn_sets_action_centric.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace stubborn_sets {
+class StubbornSetsActionCentric : public stubborn_sets::StubbornSets {
+    /*
+      stubborn_queue contains the operator indices of operators that
+      have been marked as stubborn but have not yet been processed
+      (i.e. more operators might need to be added to stubborn because
+      of the operators in the queue).
+    */
+    std::vector<int> stubborn_queue;
```
Comment (silvansievers) [inline src/search/pruning/stubborn_sets_action_centric.h:14]: I would add a newline between members and methods.

File: src/search/pruning/stubborn_sets_action_centric.cc [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+// Relies on both fact sets being sorted by variable.
+static bool contain_conflicting_fact(const vector<FactPair> &facts1,
```
Comment (silvansievers) [inline src/search/pruning/stubborn_sets_action_centric.cc:10]: I would move this function out of the way of the all of the class's definitions, i.e., above the constructor.

**Pull request closed without merging**